base/show.jl

```diff
@@ -3114,17 +3114,23 @@ end
 summary(io::IO, f::Function) = show(io, MIME"text/plain"(), f)
 
 """
-    showarg(io::IO, x, toplevel)
+    Base.showarg(io::IO, x, toplevel)
 
-Show `x` as if it were an argument to a function. This function is
-used by [`summary`](@ref) to display type information in terms of sequences of
-function calls on objects. `toplevel` is `true` if this is
-the direct call from `summary` and `false` for nested (recursive) calls.
+Show the quasi-type of `x` where quasi-type is the type of `x` or an expression (possibly
+containing quasi-types) that would generate an object of the same type as `x`. The shorter
+of these two options is typically used.
 
-The fallback definition is to print `x` as "::\\\$(typeof(x))",
-representing argument `x` in terms of its type. (The double-colon is
-omitted if `toplevel=true`.) However, you can
-specialize this function for specific types to customize printing.
+This function is used by summary to display type information in terms of sequences of
+function calls on objects.
+
+Show a leading `::` if `toplevel` is `false` and showing a type. `toplevel` is `true` if
+this is the direct call from `summary` and `false` for nested (recursive) calls.
+
+The fallback definition is to print x as "::\\\$(typeof(x))", representing argument x in terms
+of its type. (The double-colon is omitted if toplevel=true.) However, you can specialize
```

Comment:
```suggestion
of its type. (The double-colon is omitted if `toplevel=true`.) However, you can specialize
```
This is partially redundant with the previous description of `toplevel`. I'm surely biased and thus unreliable (I wrote this initially, and docs always make sense to their author no matter how confusing they really are), but I find this somewhat clearer than the more complete description you added above. Would it work to add more detail here and remove the one above?

I don't mean to portray this redundancy as a must-fix; I could certain live with merging this as-is.

Reply:
I agree that the previous description is more clear. 

Deleting the parenthetical clause entirely and without amendment would make the prior sentence technically false. Would you like it better if we delete the parenthetical and replace
> print x as "::$(typeof(x))"

with

> print x as "::$(typeof(x))" or "$(typeof(x))"

?